lib/logstash/outputs/redis.rb

```diff
@@ -185,7 +188,13 @@ def congestion_check(key)
     return if @congestion_threshold == 0
     if (Time.now.to_i - @congestion_check_times[key]) >= @congestion_interval # Check congestion only if enough time has passed since last check.
       while @redis.llen(key) > @congestion_threshold # Don't push event to Redis key which has reached @congestion_threshold.
-        @logger.warn? and @logger.warn("Redis key size has hit a congestion threshold #{@congestion_threshold} suspending output for #{@congestion_interval} seconds")
+        if @shuffle_hosts
```

Comment:
Why would we do this only if shuffle_hosts is set? Did you mean `if @congestion_switch` ?